### DISCUSSION TITLE
Kamp?

### DISCUSSION THREAD

**Discussion by b0red04** [bassamanator/Sovol-SV06-firmware#47] [Ideas]
Any plans to add Kamp?

**Comment by bassamanator** [comment]
I started testing KAMP on this repo more than a month ago, see `adaptive` branch. Unfortunately, KAMP is not production ready: has issues, and hasn't been updated in like 2 months. An advanced user can handle these shortcomings, however, a beginner will run into issues. It's understandable, because there's never enough time, and that dev, such as myself, do all this open source work for free.

I'm just waiting for the next release of KAMP. Not entirely sure how I'll manage things, will it live on a separate branch or will it be part of the main repo? We shall see.

**Reply by armishmajerj** [reply]
I'm curious what issues you were having? I actually found that branch and used it to start mine and I know I made some tweaks and had to comment out the old PURGE_LINE command so it didn't do both. I've been using it for my prints and it's been a big help. 

I will say the only issue I have is sometimes it seems like klipper/mainsail doesn't read the object sizes right and seems to just remeasure the whole bed before printing. Not the worst thing in the world just something I noticed. It'll even say how many objects it detected just does a whole bed mesh. Very odd

**Reply by bassamanator** [reply]
>I actually found that branch and used it to start mine and I know I made some tweaks and had to comment out the old PURGE_LINE command so it didn't do both.

This is why I love Klipper. A user that's willing to put in a little work, is able to make important and necessary changes to the functioning of their printer relatively easily.

>I will say the only issue I have is sometimes it seems like klipper/mainsail doesn't read the object sizes right and seems to just remeasure the whole bed before printing.

To me that issue is unacceptable. I'm trying to provide a perfect experience to my users.

The issue that I encountered was the speed at which the purge line was printed at. A deal breaker for me? No. A deal breaker for my repo? Absolutely. There are other issues as well, just check the issues section. I considered creating a pull request to fix that issue, but there is a fix in the works already.
